src/Sql/Sql/Elastic Jobs/Cmdlet/Job/NewAzureSqlElasticJob.cs

```diff
@@ -0,0 +1,267 @@
+        [ValidateNotNullOrEmpty]
+        public override string AgentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job name
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = DefaultParameterSet, Position = 3, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = AgentDefaultRunOnceParameterSet, Position = 3, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = AgentDefaultRecurringParameterSet, Position = 3, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = InputObjectParameterSet, Position = 1, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentObjectRunOnceParameterSet, Position = 1, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentObjectRecurringParameterSet, Position = 1, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, Position = 1, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentResourceIdRunOnceParameterSet, Position = 1, HelpMessage = "The job name")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentResourceIdRecurringParameterSet, Position = 1, HelpMessage = "The job name")]
+        [ValidateNotNullOrEmpty]
+        [Alias("JobName")]
+        public override string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the switch parameter run once
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = AgentDefaultRunOnceParameterSet, HelpMessage = "The flag to indicate job will be run once")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentObjectRunOnceParameterSet, Position = 2, HelpMessage = "The flag to indicate job will be run once")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentResourceIdRunOnceParameterSet, Position = 2, HelpMessage = "The flag to indicate job will be run once")]
+        public SwitchParameter RunOnce { get; set; }
+
+        /// <summary>
+        /// Get or sets the job schedule interval type
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = AgentDefaultRecurringParameterSet,
+            HelpMessage = "The recurring schedule interval type - Can be Minute, Hour, Day, Week, Month")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentObjectRecurringParameterSet, Position = 2,
+            HelpMessage = "The recurring schedule interval type - Can be Minute, Hour, Day, Week, Month")]
+        [Parameter(Mandatory = true, ParameterSetName = ParentResourceIdRecurringParameterSet, Position = 2,
+            HelpMessage = "The recurring schedule interval type - Can be Minute, Hour, Day, Week, Month")]
+        public IntervalTypes? IntervalType { get; set; }
```

Comment:
@johnpaulkee rather than using enums as the type of a parameter, we advise to use a string with the `PSArgumentCompleter` attribute -- this allows the API to updated with additional values and the user can immediately provide them to this cmdlet without having to wait for another version of the module to be released with the new value(s) added to the enum.

Reply:
Updated